inc/composer/class-command.php

```diff
@@ -671,7 +759,75 @@ protected function get_project_subdomain() : string {
 	 *
 	 * @return boolean
 	 */
-	protected function is_linux() : bool {
+	public static function is_linux() : bool {
 		return in_array( php_uname( 's' ), [ 'BSD', 'Linux', 'Solaris', 'Unknown' ], true );
 	}
+
+	/**
+	 * Check if the current host operating system is Mac OS.
+	 *
+	 * @return boolean
+	 */
+	public static function is_macos() : bool {
+		return php_uname( 's' ) === 'Darwin';
+	}
+
+	/**
+	 * Check for docker-sync file.
+	 *
+	 * @return boolean
+	 */
+	protected function has_docker_sync_file() : bool {
+		return file_exists( getcwd() . DIRECTORY_SEPARATOR . 'vendor' . DIRECTORY_SEPARATOR . 'docker-sync.yml' );
+	}
+
+	/**
+	 * Check if Mutagen is installed.
+	 *
+	 * @return boolean
+	 */
+	protected function is_mutagen_installed() : bool {
+		static $is_installed;
+		if ( $is_installed !== null ) {
+			return $is_installed;
+		}
+		if ( self::is_linux() || self::is_macos() ) {
+			$is_installed = ! empty( shell_exec( 'which mutagen' ) );
+		} else {
+			$is_installed = ! empty( shell_exec( 'where mutagen' ) );
+		}
+		// Create stub mutagen config file.
+		if ( $is_installed ) {
```

Comment:
This is a `is_*` function, it should not have side effects like this IMO.

Reply:
Yeah I could move that elsewhere 👍